internal/iam/repository_role_grant_test.go

```diff
@@ -594,19 +596,19 @@ func TestGrantsForUser(t *testing.T) {
 		WithSkipDefaultRoleCreation(true),
 	)
 	noGrantOrg1Role := TestRole(t, conn, noGrantOrg1.PublicId)
-	TestRoleGrant(t, conn, noGrantOrg1Role.PublicId, "ids=o_noGrantOrg1;actions=*")
+	TestRoleGrant(t, conn, noGrantOrg1Role.PublicId, "ids=*;type=scope;actions=*")
 	noGrantProj1Role := TestRole(t, conn, noGrantProj1.PublicId)
-	TestRoleGrant(t, conn, noGrantProj1Role.PublicId, "ids=p_noGrantProj1;actions=*")
+	TestRoleGrant(t, conn, noGrantProj1Role.PublicId, "ids=*;type=*;actions=*")
```

Comment:
I don't follow a lot of these changes to the grant strings. It looks to me like the previous grant strings weren't necessarily valid? But the new grant string doesn't seem like it would be equivalent since it is going from a single id to `*`. Can you explain the reason for these changes? Maybe it is just that this test has significant changes and I should just read the new test instead of looking at the diff.

Reply:
This very particular change is because I decided for the org grant that it was easier to understand what it was doing by making it clear it was giving scope grants, and for the project it was a bogus grant anyways because there are no scopes in a project. So instead I'm granting everything in that project to show that nothing ends up being supported down the line.

I do think you should only focus on the end result -- I was doing cleanup along the way to make it clearer that the test is demonstrating what we want with more real-world grants. Don't worry about what was there before, the code that feeds the test has changed drastically and the tests have massively changed and expanded.